libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -301,6 +301,7 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 		using Location = VariableDeclaration::Location;
 		Location varLoc = _variable.referenceLocation();
 		DataLocation typeLoc = DataLocation::Memory;
+		bool const v050 = _variable.sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
```

Comment:
The next release will be version 0.5.0, so there's no need [anymore] to optionally check for this, is it?

Reply:
Yes, since this is a breaking change, the warning should be turned into an error.